examples/dt/uni01alpha/README.md

```diff
@@ -0,0 +1,259 @@
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+    service: cinder
+  name: 90-master-cinder-enable-iscsid
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+        - enabled: true
+          name: iscsid.service
+```
+
+#### Multipath
+
+It is assumed *multipath* services are enabled in all nodes particpating in the
+Red Hat OpenShift cluster. If not, a `MachineConfig` like the one below must be
+applied. The node requires a *reboot* for the changes to be applied.
+
+```YAML
+---
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+    service: cinder
+  name: 91-master-cinder-enable-multipathd
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - path: /etc/multipath.conf
+          overwrite: false
+          # Mode must be decimal, this is 0600
+          mode: 384
+          user:
+            name: root
+          group:
+            name: root
+          contents:
+            # Source can be a http, https, tftp, s3, gs, or data as defined in rfc2397.
+            # This is the rfc2397 text/plain string format
+            source: data:,defaults%20%7B%0A%20%20user_friendly_names%20no%0A%20%20recheck_wwid%20yes%0A%20%20skip_kpartx%20yes%0A%20%20find_multipaths%20yes%0A%7D%0A%0Ablacklist%20%7B%0A%7D
+    systemd:
+      units:
+      - enabled: true
+        name: multipathd.service
+```
+
+The plain text contents of the multipath configuration file is
+
+```conf
+defaults {
+  user_friendly_names    no
+  recheck_wwid    yes
+  skip_kpartx    yes
+  find_multipaths  yes
+}
+blacklist {
+}
+```
+
+#### Cinder backend - LVM
+
+It is assumed that worker nodes or the master nodes have extra disks and there
+exists a logical volume with the name *cinder_volumes*. If not, a
+`MachineConfig` like the one below must be applied.
+
+```YAML
+---
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: 92-create-logical-volume
+  labels:
+    machineconfiguration.openshift.io/role: master
+    service: cinder
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      disks:
+        - device: "/dev/sdb"
+          wipeTable: true
+        - device: "/dev/sdc"
+          wipeTable: true
+      files:
+        - path: /usr/local/bin/lv-cinder-volumes.sh
+          overwrite: false
+          mode: 493
+          user:
+            name: root
+          group:
+            name: root
+          contents:
+            source: data:text/plain;charset=utf-8;base64,{{ _replaced_ }}
+    systemd:
+      units:
+        - name: lv-cinder-volumes.service
+          enabled: true
+          contents: |
+            [Unit]
+            Description=Create logical volume with name cinder-volumes.
+            After=var.mount systemd-udev-settle.service
+
+            [Service]
```

Comment:
You need to re-sync the systemd service settings with https://github.com/openstack-k8s-operators/cinder-operator/blob/main/config/samples/backends/bases/lvm/lvm.yaml

One of the important findings is the MachineConfig needs to be applied to all of the OCP nodes (due to restrictions on machine pools), and yet only a single node (the one with the "openstack.org/cinder-lvm" label should actually create the cinder-volumes VG. This, in turn, means one of the nodes must be preselected by applying that label.

Reply:
I had tweaked a bit as `lvm2-monitor` services was exiting. As part of deploying OCP, we are adding extra disks to the platform. Currently, the workflow adds two extra disks of size 10G each. Using the specified `machineconfig` we are configuring them.

I will need to insert labeling of a OCP node. 

Would it be a problem if we have `cinder-volumes` volume group in all Nodes and a OpenShift node has the label `openstack.org/cinder-lvm: ""`?

Reply:
Thanks for the confirmation and information on the number of disks.

I'll make the (disk) config changes to reflect one disk and 30G space.